src/main/java/edu/kit/compiler/codegen/PhiRemover.java

```diff
@@ -0,0 +1,72 @@
+package edu.kit.compiler.codegen;
+
+import com.sun.jna.Pointer;
+import edu.kit.compiler.codegen.PhiPermutationSolver.PhiAssignment;
+import edu.kit.compiler.codegen.PhiPermutationSolver.PhiSourceNodeRegister;
+import edu.kit.compiler.intermediate_lang.Instruction;
+import edu.kit.compiler.intermediate_lang.RegisterSize;
+import firm.Mode;
+import firm.nodes.Block;
+import firm.nodes.Node;
+import firm.nodes.NodeVisitor;
+
+import java.util.List;
+import java.util.Optional;
+
+public class PhiRemover {
+
+    private final BasicBlocks basicBlocks;
+    private static final Node nullNode = new Node(Pointer.createConstant(1)) {
+        @Override
+        public void accept(NodeVisitor nodeVisitor) {
+            // do nothing
+        }
+    };
+
+    public PhiRemover(BasicBlocks basicBlocks) {
+        this.basicBlocks = basicBlocks;
+    }
+
+    /**
+     * transforms this list of blocks into a new one without any phi instructions
+     *
+     * @return list of basic blocks without phis
+     */
+    public BasicBlocks removePhis() {
```

Comment:
I would have expected your code to receive an instance of `BasicBlocks` and return some kind of `Map<Integer, List<Instruction>>` (which is what we need for register allocation).